kge/job/train.py

```diff
@@ -538,6 +547,21 @@ def __init__(self, config, dataset, parent_job=None):
                     )
                 )
 
+        #' for each query type: list of queries
+        self.queries = {}
+
+        #' for each query type: list of all labels (concatenated across queries)
+        self.labels = {}
+
+        #' for each query type: list of starting offset of labels in self.labels. The
+        #' labels for the i-th query of query_type are in labels[query_type] in range
+        #' label_offsets[query_type][i]:label_offsets[query_type][i+1]
+        self.label_offsets = {}
+
+        #' for each query type (ordered as in self.query_types), index right after last
+        #' example of that type in the list of all examples
+        self.query_end_index = {}
```

Comment:
Why was this moved?

Reply:
Just for consistency. Usually in our codebase we always declare class instance variables in the init(). I can revert if you want.

Reply:
I see. I suggest to move back for readability. Alternatively, set to `None` here to make clear that this is not correctly initialized.